runtime/src/main/java/com/flipkart/flux/representation/ClientElbPersistenceService.java

```diff
@@ -0,0 +1,159 @@
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.transaction.Transactional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <code>ClientElbPersistenceService</code> class does persistence related operations in memory
+ * before storing, updating, reading or deleting objects in DB. It is also responsible for maintaining
+ * an in-memory cache(Google Guava: https://github.com/google/guava/wiki) of clientElb entries for faster query results.
+ *
+ * @author akif.khan
+ */
+@Singleton
+public class ClientElbPersistenceService {
+
+    private ClientElbDAO clientElbDAO;
+
+    private LoadingCache<String, ClientElb> clientElbCache;
+
+    private Integer MAX_CACHE_SIZE;
+
+    private static final Logger logger = LoggerFactory.getLogger(ClientElbPersistenceService.class);
+
+    @Inject
+    public ClientElbPersistenceService(ClientElbDAO clientElbDAO,
+                                       @Named("elbCache.maxSize") Integer MAX_CACHE_SIZE) {
+        this.clientElbDAO = clientElbDAO;
+        this.MAX_CACHE_SIZE = MAX_CACHE_SIZE;
+        this.clientElbCache = CacheBuilder.newBuilder().maximumSize(MAX_CACHE_SIZE)
+                .expireAfterAccess(1, TimeUnit.DAYS)
+                .build(new CacheLoader<String, ClientElb>() {
+                    @Override
+                    public ClientElb load(String clientElbId) throws Exception {
+                        //Call ClientElbDAO layer, if entry not found in cache
+                        return clientElbDAO.findById(clientElbId);
+                    }
+                });
+    }
+
+    public long getClientElbCacheSize() {
+        return this.clientElbCache.size();
+    }
+
+    public boolean clientElbCacheContainsKey(String key) {
+        return this.clientElbCache.asMap().containsKey(key);
+    }
+
+    public String getClientElbCacheUrl(String id) {
+        String clientElbUrl = null;
+        try {
+            clientElbUrl = this.clientElbCache.get(id).getElbUrl();
+        } catch (ExecutionException e) {
+            logger.error("ClientElbCache entry not found in both DB and cache {} {}", e.getMessage(), e.getStackTrace());
+        }
+        return clientElbUrl;
+    }
+
+    /**
+     * Converts {@link ClientElbDefinition} to domain object {@link ClientElb}
+     *
+     * @param clientElbDefinition
+     * @return clientElb domain object
+     */
+    public ClientElb convertClientElbDefinitionToClientElb(ClientElbDefinition clientElbDefinition) {
+        return new ClientElb(clientElbDefinition.getId(), clientElbDefinition.getElbUrl());
+    }
+
+    /**
+     * Persists the ClientElb details object in DB.
+     *
+     * @param clientElbDefinition
+     * @return created clientElb
+     */
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public ClientElb persistClientElb(String clientId, ClientElbDefinition clientElbDefinition) {
```

Comment:
Why do these functions need to be transactional, if the underlying functions are transactions already.  Also these functions should be part of the api layer which accesss DAO .

Reply:
Removed @Transactional from these Service layer CRUD methods. Earlier, before introducing
Guava Cache, I defined a custom LRU cache where these methods were @Transactional, not required anymore. 
findById, update and delete need to be a part of this cache service layer because in each of these cases, cache needs to read, refresh/update and delete entry respectively.
We can afford to move create operation directly to API layer, however for the modularity of the code, I think we should let it pass via this cache service layer itself.